etl/grapher_helpers.py

```diff
@@ -286,3 +287,143 @@ def join_sources(sources: List[catalog.meta.Source]) -> catalog.meta.Source:
             meta[key] = sep.join(keys)
 
     return catalog.meta.Source(**meta)
+
+
+# TODO: The following function and join_sources are redundant. Merge the functionality of both into one function.
+def combine_metadata_sources(metadata: catalog.DatasetMeta) -> catalog.DatasetMeta:
```

Comment:
Can you then remove `join_sources`, replace its use in `population_density` and find/replace all its mentions?

Reply:
ok, I'm already working in a different sub-branch, I'll do it there.